nestkernel/proxynode.h

```diff
@@ -104,35 +104,35 @@ class proxynode : public Node
    * problems with dictionary entry checking
    */
   void
-  set_status( const DictionaryDatum& )
+  set_status( const DictionaryDatum& ) override
   {
     assert( false );
   }
 
-  bool is_proxy() const;
+  bool is_proxy() const override;
 
-  thread
-  get_thread() const
+  static thread
+  get_thread()
```

Comment:
A static function is (ever so slightly) more performant than a regular one, as a static function does not receive the implicit _this_ parameter, which is one less pointer to be sent around.

Reply:
I would argue that it increases code complexity a little (programmers need to keep in mind that it is static), so unless there is a demonstrable gain in performance, I'd advise against `static` here.

But this makes me wonder if there is not another problem: `Node::get_thread()` is *not* `virtual`, but `proxynode`, which derives from `Node`, defines its own `get_thread()`. Can this be right?

Reply:
Well, for programmers nothing really changes, as the function can still be called just like before. I can just make it `const` though, as well, the performance gain would be neglectable anyway.

I had a discussion with @med-ayssar regarding this issue. We came to the conclusion that `virtual` is indeed not needed in many cases where classes derive from `Node` (e.g. `set_spiketime`). However, in this case we need the virtual specifier, as the `get_thread` function will indeed be called on derived classes from the `Node::` interface and therefore the "overriden" implementation would not be used in these cases.
@jougs Can you confirm this?

Reply:
True, that helps!

Reply:
Thanks for the details, @jougs! I have thought it through once more, and the situation is thus:
- Methods in `Node` that deal only with fields in the `Node` base class, such as `get_thread()`, are plain (not virtual) methods, because they will work correctly even when used through a base class pointer or reference. 
- Having these methods non-virtual makes them faster.
- The only exception here is `proxynode`, on which `get_thread()` must never be called.
- `proxynode` defines its own `get_thread()`, thus hiding the one in `Node`. This means that
    - if we call `get_thread()` through `proxynode*`, we trigger the assertion
    - if we call it trough `Node*` or `ArchivingNode*`, we call `Node::get_thread()`. Are we sure this is right?
- No matter what, I would argue against the `static`—no gain and just potential confusion.

Reply:
* That is not guaranteed, but is valid in this case. It might be possible that an implementation in a derived class does something different with the same fields of the base class compared to the implementation in the base class.
* Exactly. Static > Non-virtual const > Non-virtual > Virtual. Having _any_ virtual function in class will have a general impact on performance by the way, as having at least one virtual function will add a vtable to instances of the class with `sizeof(pointer)` (usually 4 or 8 bytes), which might have an impact on caching performance.
* The `get_thread` of the `proxynode` could very well also be deleted, the reason for its existence is only to see that the function does not exist for a reason (right?).
* Correct. Just tested this and yes, I get these results as well.
* Alright, will change the `static` back to `const`.

Reply:
Thanks!